fork_choice_store/src/state_cache_processor.rs

```diff
@@ -86,6 +87,27 @@ impl<P: Preset> StateCacheProcessor<P> {
             store,
             block_root,
             slot,
+            ALLOWED_EMPTY_SLOTS_MULTIPLIER,
+            should_print_slot_processing_warning(store),
+        )
+    }
+
+    // Processing long slot distances are expensive and should not be allowed.
+    // (especially when the processed state is needed for things like attestations
+    // - that could lead to excessive mem and CPU usage and result in DoS).
+    // The exception is block sync - which should be allowed, because it's
+    // the only way for the chain to progress in long non-finality events.
```

Comment:
"in long non-finality events" should be replaced with "long periods without blocks"